### PR TITLE
feat: List available badges on Home & Browse Pages

### PR DESCRIPTION
**Pull request by mgorsk1**
### Summary of Changes

Introduces Available badges on landing page and Browse badges in Browse page.

Available badges component on frontend is rendered only if /badges returns non-empty list of badges
So far all badges are shown under Available badges header - we could go with grouping badges into badge categories (header would contain name of category and list of badges would show badges under this category)

This is follow-up to RFC: amundsen-io/rfcs#44

---
This PR introduces following changes to UI:

- **Home Page:**
image

- **Browse Page:**
image

### Tests

- done

### Documentation

- n/a

### CheckList
Make sure you have checked **all** steps below to ensure a timely review.
- [x] PR title addresses the issue accurately and concisely. Example: "Updates the version of Flask to v1.0.2"
    - In case you are adding a dependency, check if the license complies with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
- [x] PR includes a summary of changes.
- [x] PR adds unit tests, updates existing unit tests, __OR__ documents why no test additions or modifications are needed.
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - All the public functions and the classes in the PR contain docstrings that explain what it does

### CODE REVIEW

**Comment by youngyjd** [comment]
It looks like both column badges and table badges are showing in the homepage. I think it maybe better to just display table badges on homepage? clicking on a column badge which takes me to a search result page literally without any table

**Comment by mgorsk1** [comment]
that's a good point @youngyjd! I think the easiest way would be to do it in the metadata proxy - show only badges connected to tables (modify cypher query). unfortunately badge is generic type and there is no other distinction between table/column badges.

File: frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+};
+
+const BadgeBrowseListLong: React.FC<BadgeBrowseListProps> = ({
+  badges,
+}: BadgeBrowseListProps) => (
+  <article className="badges-browse-section badges-browse-section-long">
+    <h1 className="header-title">{BROWSE_BADGES_TITLE}</h1>
+    <hr className="header-hr" />
+    <label className="section-label">
+      <span className="section-title title-2">{AVAILABLE_BADGES_TITLE}</span>
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/index.tsx:44]: The 'title-X' class is deprecated, please add the stiles to 'section-title' with an extend or use classes like text-body-w2

File: frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/styles.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
+// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+@import 'variables';
+@import 'typography';
+
+.badges-browse-section {
+  margin: 32px 0;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/styles.scss:8]: Let's try to make this values a multiple of the grid or use the spacer variables

File: frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/styles.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    display: block;
+    font-weight: $font-weight-body-regular;
+    margin-top: $spacer-3;
+    margin-bottom: $spacer-2;
+  }
+}
+
+.badges-browse-section-short {
+  .available-badges-header-title {
+     @extend %text-title-w1;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeBrowseList/styles.scss:20]: This is the styling we want

File: frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss [hunk 4]
```diff
@@ -11,11 +11,96 @@
   margin: 0 0 0 $spacer-1;
 }
 
+.label {// Copyright Contributors to the Amundsen project.
+// SPDX-License-Identifier: Apache-2.0
+
+@import 'variables';
+
+.flag {
+  border-radius: 5px;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss:20]: Let's add a variable for this

File: frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss [hunk 5]
```diff
@@ -11,11 +11,96 @@
...
+  }
+
+  .label {
+    padding: 0 0;
+  }
+
+  [class^='badge-overlay-'] {
+    font-size: $font-size-base;
+    text-align: center;
+    padding: 0.2em 0.6em 0.3em;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss:44]: Again, we are not using ems in Amundsen for the moment

File: frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss [hunk 6]
```diff
@@ -11,11 +11,96 @@
...
+
+    &:active {
+      background-color: rgba(
+        $color: $badge-overlay,
+        $alpha: $badge-pressed-dark
+      );
+    }
+  }
+}
+  border-radius: 10px;
```
Comment (Golodhros) [inline frontend/amundsen_application/static/js/components/Badges/BadgeList/styles.scss:100]: variable too

**Pull request merged**